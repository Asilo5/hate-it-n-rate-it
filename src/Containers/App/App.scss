* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.App {
  font-family: 'Gothic A1', sans-serif;
}

#blockColorblindContent {
  display: none;
}

