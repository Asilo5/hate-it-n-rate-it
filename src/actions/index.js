export const addMoviesData = movies => ({
  type: 'ADD_MOVIES_DATA',
  movies
})

export const setUser = user => ({
  type: 'SET_USER',
  user
})

export const hasError = (error) => ({
  type: 'HAS_ERROR',
  error
})

