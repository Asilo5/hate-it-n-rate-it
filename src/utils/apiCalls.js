export const fetchUser = async (email, password) => {
  const userInfo = {email, password};
  const options = {
    method: 'POST',
    body: JSON.stringify(userInfo),
    headers: {'Content-Type': 'application/json'}
  }
  const response = await fetch('https://rancid-tomatillos.herokuapp.com/api/v1/login', options);
  if (!response.ok) {
    const error = await response.json();
    throw new Error(error.message);
  }
  const user = await response.json();
  return user;
}

export const fetchRatings = async () => {
  const response = await fetch('https://rancid-tomatillos.herokuapp.com/api/v1/users/5/ratings');
  if (!response.ok) {
    const error = await response.json();
    throw new Error(error.message);
  }
  const movieRatings = await response.json();
  console.log(movieRatings);
  return movieRatings;
}