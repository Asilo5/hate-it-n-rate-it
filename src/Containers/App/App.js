import React, { Component } from 'react';
import './App.scss';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';
import MoviesContainer from '../MoviesContainer/MoviesContainer';
import { addMoviesData } from '../../actions/index';
import Login from '../Login/Login';
import Header from '../../Components/Header/Header';
import NavBar from '../../Components/NavBar/NavBar';
import Movie from '../Movie/Movie';


export class App extends Component {
  constructor() {
    super();
  }

  componentDidMount() {
    fetch('https://rancid-tomatillos.herokuapp.com/api/v1/movies')
      .then(response => response.json())
      .then(data => this.props.addMoviesData(data))
  }

  render() {
    const { movies } = this.props;
    return (
      <main className="App">
        <Header />
        <NavBar />
        <Route exact path='/' component={MoviesContainer} /> 
        <Route exact path='/login' component={Login} />
        <Route path='/movies/:id' render={ ({ match }) => {
          const { id } = match.params;
          
          const foundFilm = movies.find((movie) => movie.id === parseInt(id));

          return <Movie {...foundFilm}/>
 
        }} />
      </main>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  addMoviesData: movies => dispatch( addMoviesData(movies))
})

const mapStateToProps = ({ movies }) => ({
   movies
})


export default connect(mapStateToProps, mapDispatchToProps)(App);
