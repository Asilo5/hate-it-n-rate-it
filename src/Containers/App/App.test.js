import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { getMovies } from '../../utils/apiCalls'; 
import { addMoviesData } from '../../actions';
jest.mock('../../utils/apiCalls');


describe('App Component', () => {

  describe('App', () => {

    let wrapper;
    let mockMovies = [
      {
        id: 1,
        title: "Jumanji: The Next Level",
        poster_path: "https://image.tmdb.org/t/p/original//l4iknLOenijaB85Zyb5SxH1gGz8.jpg",
        backdrop_path: "https://image.tmdb.org/t/p/original//zTxHf9iIOCqRbxvl8W5QYKrsMLq.jpg",
        release_date: "2019-12-04",
        overview: "In Jumanji: The Next Level, the gang is back but the game has changed. As they return to rescue one of their own, the players will have to brave parts unknown from arid deserts to snowy mountains, to escape the world's most dangerous game.",
        average_rating: 6
      }
    ];
  
    beforeEach(() => {
      getMovies.mockImplementation(() => {
        return Promise.resolve(mockMovies)
      });
  
      wrapper = shallow(<App movies={mockMovies}
                             addMoviesData={jest.fn()} />);
   
    })
  
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })
  
    it('should call getMovies after component is mounted', () => {
      expect(getMovies).toHaveBeenCalled();
    })
  
  })

  describe('mapDispatchToProps', () => {
     let mockMovies = [
      {
        id: 1,
        title: "Jumanji: The Next Level",
        poster_path: "https://image.tmdb.org/t/p/original//l4iknLOenijaB85Zyb5SxH1gGz8.jpg",
        backdrop_path: "https://image.tmdb.org/t/p/original//zTxHf9iIOCqRbxvl8W5QYKrsMLq.jpg",
        release_date: "2019-12-04",
        overview: "In Jumanji: The Next Level, the gang is back but the game has changed. As they return to rescue one of their own, the players will have to brave parts unknown from arid deserts to snowy mountains, to escape the world's most dangerous game.",
        average_rating: 6
      }
    ];

    const mockDispatch = jest.fn();

    it('should dispatch setUser when handleSubmit', () => {
      const actionToDispatch = addMoviesData(mockMovies);
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.addMoviesData(mockMovies);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  })

  describe('mapStateToProps', () => {
     const mockState = { 
        movies : [
          {
            id: 1,
            title: "Jumanji: The Next Level",
            poster_path: "https://image.tmdb.org/t/p/original//l4iknLOenijaB85Zyb5SxH1gGz8.jpg",
            backdrop_path: "https://image.tmdb.org/t/p/original//zTxHf9iIOCqRbxvl8W5QYKrsMLq.jpg",
            release_date: "2019-12-04",
            overview: "In Jumanji: The Next Level, the gang is back but the game has changed. As they return to rescue one of their own, the players will have to brave parts unknown from arid deserts to snowy mountains, to escape the world's most dangerous game.",
            average_rating: 6.222222222222222
            }
        ]
     };

    const expected = {
      movies : [
        {
          id: 1,
          title: "Jumanji: The Next Level",
          poster_path: "https://image.tmdb.org/t/p/original//l4iknLOenijaB85Zyb5SxH1gGz8.jpg",
          backdrop_path: "https://image.tmdb.org/t/p/original//zTxHf9iIOCqRbxvl8W5QYKrsMLq.jpg",
          release_date: "2019-12-04",
          overview: "In Jumanji: The Next Level, the gang is back but the game has changed. As they return to rescue one of their own, the players will have to brave parts unknown from arid deserts to snowy mountains, to escape the world's most dangerous game.",
          average_rating: 6.222222222222222
          }
      ]
    };

    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  })

})
