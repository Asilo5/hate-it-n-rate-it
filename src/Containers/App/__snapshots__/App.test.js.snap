// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<main
  className="App"
>
  <Connect(Header) />
  <Connect(NavBar) />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(MoviesContainer)",
        "type": [Function],
      }
    }
    exact={true}
    path="/"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Login)",
        "type": [Function],
      }
    }
    exact={true}
    path="/login"
  />
  <Route
    path="/movies/:id"
    render={[Function]}
  />
</main>
`;
